storage/raft_transport.go

```diff
@@ -255,9 +265,11 @@ func (t *RaftTransport) processQueue(nodeID roachpb.NodeID) {
 			}
 			return
 		case req := <-ch:
+			wg.Add(1)
```

Comment:
I think you only need to do the `wg.{Add,Done}` dance when `req.Message.Type == raftpb.MsgSnap`. In the other branch it is impossible for the `defer` to be running concurrently with `restStream.Send`.